### PR TITLE
Add validating webhook

### PR DESCRIPTION
**Pull request by erkanerol**
This PR
- update existing resources by using v5.0.1 of [upstream repo](https://github.com/kubernetes-csi/external-snapshotter/releases/tag/v5.0.1)
- adds validatingwebhook for `volumesnapshots` and `volumesnapshotcontents`
- adds `kustomize` support to template upstream manifests

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

**Review by kopiczko — COMMENTED** [review]
I didn't read the manifests

**Comment by erkanerol** [comment]
@kopiczko is there a problem in this PR?

**Comment by erkanerol** [comment]
@kopiczko I updated this PR as you want.

File: hack/prepare-helmchart.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+set -x 
+
+# ${1} is the given application-name from make (csi-external-snapshotter-app)
+
+# as we apply the crd via configmap, name must be stripped to be RFC 1123 conform
+# (lower case alphanumeric characters or '-', and must start and end with an alphanumeric character)
```
Comment (kopiczko) [inline hack/prepare-helmchart.sh:8]: Do we apply them via configmap?

Comment (erkanerol) [reply hack/prepare-helmchart.sh:8]: yes. with crd-install job.

Comment (kopiczko) [reply hack/prepare-helmchart.sh:8]: Where is this job defined?

Comment (erkanerol) [reply hack/prepare-helmchart.sh:8]: It is already a part of the main branch now. This PR just adds a webhook part and uses kustomization to generate manifests.  https://github.com/giantswarm/csi-external-snapshotter-app/tree/main/helm/csi-external-snapshotter-app/templates/crd-install

**Review by kopiczko — APPROVED** [review]

**Pull request merged**